src/Provider/AbstractProvider.php

```diff
@@ -442,36 +471,25 @@ public function getAccessToken($grant, array $options = [])
 
 
     /**
-     * Get a request instance.
+     * Get a request instance, optionally authenticated using an access token.
      *
      * Creates a PSR-7 compatible request instance that can be modified.
-     * The request is not automatically authenticated.
      *
      * @param  string $method
      * @param  string $url
+     * @param  AccessToken|string $token
      * @param  array  $options Any of "headers", "body", and "protocolVersion".
      * @return RequestInterface
      */
-    public function getRequest($method, $url, array $options = [])
+    public function getRequest($method, $url, $token = null, array $options = [])
```

Comment:
I don't really like this. In my opinion having separate public methods for authenticated and non-authenticated requests is valuable. Having to pass a `null` in the middle of a method call is a good sign that separate methods are useful, and having a hard `$token` requirement for authenticated requests allows for more accurate hinting.